File: app/src/main/java/com/shafa/ali/kavir_msg/activity/SearchActivity.java
package com.shafa.ali.kavir_msg.activity;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.SearchView;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.ImageButton;

import com.shafa.ali.kavir_msg.R;
import com.shafa.ali.kavir_msg.adapters.TitleAdapter;
import com.shafa.ali.kavir_msg.interfaces.ClickListener;
import com.shafa.ali.kavir_msg.models.TiltlePostsModel;
import com.shafa.ali.kavir_msg.server.GetPostsServer;
import com.shafa.ali.kavir_msg.utility.RecyclerTouchListener;
import com.shafa.ali.kavir_msg.utility.RetrofitClientInstance;
import com.shafa.ali.kavir_msg.utility.SaveItem;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;

public class SearchActivity extends AppCompatActivity {
    private RecyclerView recyclerView;
    private SearchView searchView;
    private ImageButton backBtn;
    private List<TiltlePostsModel.PostsModel> postsModelList;
    private FrameLayout searchFram;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);
        recyclerView =(RecyclerView)findViewById(R.id.search_result_rc);
        searchView = (SearchView)findViewById(R.id.search_dt);
        backBtn = (ImageButton)findViewById(R.id.back_search_btn);
        searchFram=(FrameLayout)findViewById(R.id.frame_et);
        backBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });

        searchView.setOnSearchClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                searchFram.setBackgroundColor(Color.WHITE);
            }
        });

        searchView.setOnCloseListener(new SearchView.OnCloseListener() {
            @Override
            public boolean onClose() {
                searchFram.setBackgroundColor(Color.parseColor("#d32f2f"));
                return false;
            }
        });

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String s) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String s) {
                getsearchResult(s);
                return false;
            }
        });
        EditText searchEditText = (EditText) searchView.findViewById(android.support.v7.appcompat.R.id.search_src_text);
//        searchEditText.setTextColor(getResources().getColor(R.color.white));
        searchEditText.setTextSize(24);
//        searchEditText.setActivated(true);
        Typeface type2 = Typeface.createFromAsset(getAssets(),"fonts/sans.ttf");
        searchEditText.setTypeface(type2,Typeface.NORMAL);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);

        recyclerView.addOnItemTouchListener(new RecyclerTouchListener(this, recyclerView, new ClickListener() {
            @Override
            public void onClick(View view, int position) {
                TiltlePostsModel.PostsModel postsModel = postsModelList.get(position);
                Intent intent = new Intent(SearchActivity.this,PostActivity.class);
                intent.putExtra("postId",postsModel.getId());
                intent.putExtra("source","net");
                startActivity(intent);
            }

            @Override
            public void onLongClick(View view, int position) {

            }
        }));
    }

    private void getsearchResult(String query) {
        recyclerView.setVisibility(View.GONE);
        Retrofit retrofit = RetrofitClientInstance.getRetrofitInstance();
        GetPostsServer getPostsServer = retrofit.create(GetPostsServer.class);
        getPostsServer.getSearch(SaveItem.getItem(this,SaveItem.USER_COOKIE,""),query).enqueue(new Callback<List<TiltlePostsModel.PostsModel>>() {
            @Override
            public void onResponse(Call<List<TiltlePostsModel.PostsModel>> call, Response<List<TiltlePostsModel.PostsModel>> response) {
                if (response.isSuccessful()){
                    TitleAdapter titleAdapter = new TitleAdapter(SearchActivity.this,response.body());
                    postsModelList = response.body();
                    recyclerView.setAdapter(titleAdapter);
                    recyclerView.setVisibility(View.VISIBLE);
                }
            }

            @Override
            public void onFailure(Call<List<TiltlePostsModel.PostsModel>> call, Throwable t) {
//                Log.e("onFailure:",t.getMessage());
            }
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }
}

File: app/src/main/java/com/kavirelectronic/ali/kavir_info/activity/SearchActivity.kt
package com.kavirelectronic.ali.kavir_info.activity

import android.content.Intent
import android.graphics.Color
import android.graphics.Typeface
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.appcompat.widget.SearchView
import android.view.View
import android.widget.EditText
import android.widget.FrameLayout
import android.widget.ImageButton
import com.kavirelectronic.ali.kavir_info.R
import com.kavirelectronic.ali.kavir_info.adapters.TitleAdapter
import com.kavirelectronic.ali.kavir_info.interfaces.ClickListener
import com.kavirelectronic.ali.kavir_info.models.TiltlePostsModel.PostsModel
import com.kavirelectronic.ali.kavir_info.server.GetPostsServer
import com.kavirelectronic.ali.kavir_info.utility.RecyclerTouchListener
import com.kavirelectronic.ali.kavir_info.utility.RetrofitClientInstance
import com.kavirelectronic.ali.kavir_info.utility.SaveItem
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class SearchActivity : AppCompatActivity() {
    private var recyclerView: RecyclerView? = null
    private var searchView: SearchView? = null
    private var backBtn: ImageButton? = null
    private var postsModelList: List<PostsModel?>? = null
    private var searchFram: FrameLayout? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search)
        recyclerView = findViewById<View>(R.id.search_result_rc) as RecyclerView
        searchView = findViewById<View>(R.id.search_dt) as SearchView
        backBtn = findViewById<View>(R.id.back_search_btn) as ImageButton
        searchFram = findViewById<View>(R.id.frame_et) as FrameLayout
        backBtn!!.setOnClickListener { onBackPressed() }
        searchView!!.setOnSearchClickListener { searchFram!!.setBackgroundColor(Color.WHITE) }
        searchView!!.setOnCloseListener {
            searchFram!!.setBackgroundColor(Color.parseColor("#d32f2f"))
            false
        }
        searchView!!.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(s: String): Boolean {
                return false
            }

            override fun onQueryTextChange(s: String): Boolean {
                getsearchResult(s)
                return false
            }
        })
        val searchEditText = searchView!!.findViewById<View>(androidx.appcompat.R.id.search_src_text) as EditText
        //        searchEditText.setTextColor(getResources().getColor(R.color.white));
        searchEditText.textSize = 24f
        //        searchEditText.setActivated(true);
        val type2 = Typeface.createFromAsset(assets, "fonts/sans.ttf")
        searchEditText.setTypeface(type2, Typeface.NORMAL)
        val layoutManager: RecyclerView.LayoutManager = LinearLayoutManager(this)
        recyclerView!!.layoutManager = layoutManager
        recyclerView!!.addOnItemTouchListener(RecyclerTouchListener(this, recyclerView!!, object : ClickListener {
            override fun onClick(view: View?, position: Int) {
                val postsModel = postsModelList!![position]
                val intent = Intent(this@SearchActivity, PostActivity::class.java)
                intent.putExtra("postId", postsModel!!.id)
                intent.putExtra("source", "net")
                startActivity(intent)
            }

            override fun onLongClick(view: View?, position: Int) {
                TODO("Not yet implemented")
            }
        }))
    }

    private fun getsearchResult(query: String) {
        recyclerView!!.visibility = View.GONE
        val retrofit = RetrofitClientInstance.retrofitInstance
        val getPostsServer = retrofit!!.create(GetPostsServer::class.java)
        getPostsServer.getSearch(SaveItem.getItem(this, SaveItem.USER_COOKIE, ""), query)!!.enqueue(object : Callback<List<PostsModel?>?> {
            override fun onFailure(call: Call<List<PostsModel?>?>, t: Throwable) {
                Log.e("fail search actiity,",t.message)
            }

            override fun onResponse(call: Call<List<PostsModel?>?>, response: Response<List<PostsModel?>?>) {
                if (response.isSuccessful) {
                    val titleAdapter = TitleAdapter(this@SearchActivity, response.body())
                    postsModelList = response.body()
                    recyclerView!!.adapter = titleAdapter
                    recyclerView!!.visibility = View.VISIBLE
                }
            }

        })
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }
}